k9mail/src/test/java/com/fsck/k9/activity/MessageReferenceTest.java

```diff
@@ -54,6 +54,30 @@ public void parseIdentityStringWithFlag() throws MessagingException {
     }
 
     @Test
+    public void checkMessageReferenceWithChangedUid() {
+        MessageReference messageReferenceOne = createMessageReferenceWithFlag("account", "folder", "uid", Flag.ANSWERED);
+        MessageReference messageReferenceTwo = messageReferenceOne.withModifiedUid("---");
+
+        assertFalse(messageReferenceOne.equals(messageReferenceTwo));
+        assertEquals(messageReferenceOne.getAccountUuid(), messageReferenceTwo.getAccountUuid());
+        assertEquals(messageReferenceOne.getFolderName(), messageReferenceTwo.getFolderName());
+        assertEquals(messageReferenceOne.getFlag(), messageReferenceTwo.getFlag());
+        assertFalse(messageReferenceOne.getUid().equals(messageReferenceTwo.getUid()));
```

Comment:
I'd prefer if the contents of `messageReferenceTwo` were checked against (string) values, not the values in `messageReferenceOne`.

Missing: `assertEquals("---", messageReferenceTwo.getUid());`